src/test/kotlin/com/terraformation/backend/funder/FundingEntityServiceTest.kt

```diff
@@ -0,0 +1,263 @@
+    val actualModel = fundingEntityStore.fetchOneById(createdModel.id)
+
+    assertEquals(expected, actualRow)
+    assertEquals(0, actualModel.projects!!.size)
+  }
+
+  @Test
+  fun `create rejects duplicates by name`() {
+    assertThrows<FundingEntityExistsException> { service.create("TestFundingEntity") }
+  }
+
+  @Test
+  fun `create adds project rows`() {
+    insertOrganization()
+    val projectId1 = insertProject()
+    val projectId2 = insertProject()
+    val projectSet = setOf(projectId1, projectId2)
+
+    val inserted = service.create("New Funding Entity with Projects", projectSet)
+
+    val actual = fundingEntityStore.fetchOneById(inserted.id)
+
+    assertEquals(2, actual.projects!!.size)
+    assertEquals(projectSet, actual.projects!!.toSet())
+  }
+
+  @Test
+  fun `update throws exception when no id attached`() {
+    assertThrows<IllegalArgumentException> {
+      service.update(FundingEntitiesRow(name = "Funding Entity without Id"))
+    }
+  }
+
+  @Test
+  fun `update throws exception if user has no permission to manage funding entities`() {
+    every { user.canUpdateFundingEntities() } returns false
+
+    assertThrows<AccessDeniedException> {
+      service.update(FundingEntitiesRow(id = fundingEntityId, name = "Updated Funding Entity"))
+    }
+  }
+
+  @Test
+  fun `update throws exception when name conflict`() {
+    insertFundingEntity("Existing Funding Entity")
+
+    assertThrows<FundingEntityExistsException> {
+      service.update(FundingEntitiesRow(id = fundingEntityId, name = "Existing Funding Entity"))
+    }
+  }
+
+  @Test
+  fun `update throws exception when no matching entity`() {
+    assertThrows<FundingEntityNotFoundException> {
+      service.update(
+          FundingEntitiesRow(id = FundingEntityId(1093), name = "Missing Funding Entity"))
+    }
+  }
+
+  @Test
+  fun `update populates modifiedBy fields`() {
+    val newTime = clock.instant().plusSeconds(1000)
+    clock.instant = newTime
+
+    val newUserId = insertUser()
+
+    val updates =
+        FundingEntitiesRow(
+            id = fundingEntityId,
+            name = "New Name",
+        )
+    val expected =
+        updates.copy(
+            createdBy = user.userId,
+            createdTime = Instant.EPOCH,
+            modifiedBy = newUserId,
+            modifiedTime = newTime,
+        )
+
+    every { user.userId } returns newUserId
+
+    service.update(updates)
+
+    val actual = fundingEntitiesDao.fetchOneById(fundingEntityId)!!
+    assertEquals(expected, actual)
+  }
+
+  @Test
+  fun `update correctly adds new projects`() {
+    val row = FundingEntitiesRow(id = fundingEntityId, name = "TestFundingEntity")
+    val projectId1 = insertProject()
+
+    assertEquals(0, fundingEntityProjectsDao.fetchByFundingEntityId(fundingEntityId).size)
+
+    service.update(row, setOf(projectId1))
+
+    assertEquals(1, fundingEntityProjectsDao.fetchByFundingEntityId(fundingEntityId).size)
```

Comment:
Could use `assertTableEquals` here.

Also, though this isn't as much of an issue if you use `assertTableEquals`, when I have multiple calls to the same `assert` method in a test, especially if the arguments are of the same type, I like to make sure each one has a description so it's easier to tell at a glance which one failed.